exploration/open-close-expressions.md

```diff
@@ -220,3 +221,70 @@ Rendered as React this could become:
 _What other solutions are available?_
 _How do they compare against the requirements?_
 _What other properties they have?_
+
+### New declarations and a familiar syntax
+
+The goal of this solution is to avoid adding new sigils to the syntax.
+Instead, it leverages *keywords* as the tool for qualifying certain placeholders as open, close, or standalone,
+combined with familiar HTML-inspired syntax.
+
+This solution consists of adding:
+
+* Two new declaration types called `openclose` and `standalone`.
```

Comment:
`openclose` is weird. How about `span` (for spannables) and `tag` (for standalone tags)?

Reply:
I like the suggestions.

(I may have become become a *connoisseur* of intended ugliness at the design stage ;) )

Reply:
This part is pretty much orthogonal to the design doc objective:
> Describe the use cases and requirements for _expressions_
> that enclose parts of a pattern
> and develop a design that satisfies these needs.

If/as we should discuss packaging options to a message-local tag, that should be separate from this doc.

If we go with `{+foo}` / `{-foo}`, then something like
```
local +foo = ...
```
is probably considerable. On the other hand, with `{foo}` / `{/foo}`, a separate keyword or two is probably a right answer:
```
tag foo = ...
```

In that case, I do believe that you'll find that a single `tag` is quite sufficient, rather than needing a wholly separate one for standalone tags. We do not need to build congruence into the language at this level to guarantee that a local `{foo}` will never be used as `{foo/}`; that's properly handled via the registry.

As far as I've understood, the design docs are meant to rationalize decisions, rather than discussions. So we can have a discussion spanning more than one doc leading to two related conclusions.

Reply:
> On the other hand, with `{foo}` / `{/foo}`, a separate keyword or two is probably a right answer:
> 
> ```
> tag foo = ...
> ```
> 
> In that case, I do believe that you'll find that a single `tag` is quite sufficient, rather than needing a wholly separate one for standalone tags. We do not need to build congruence into the language at this level to guarantee that a local `{foo}` will never be used as `{foo/}`; that's properly handled via the registry.

You're right, two declarations duplicate some of the responsibilities of the registry. I originally started out with just a single declaration, but then decided that it was better to be overly specific rather than too generic for the purpose of the design process.

I'll push an update later today. `tag`? `markup`? `element`?

Reply:
Also, I think you're also right that adding declarations addresses a different set of problems. I'm okay with moving declarations to another proposal ("open/close *declarations*") and keeping this doc focused on "open/close *expressions*".

Reply:
> I'll push an update later today. `tag`? `markup`? `element`?

I'd prefer at most 5 characters, given that that's the max otherwise.

Reply:
I'd prefer `tag`, since it isn't opinionated that the tag is markup, but not so unopinionated (`element`) that it has little meaning. Also, it's short. I can add this to the list for a captive audience today.